tools/c7n_oci/c7n_oci/log.py

```diff
@@ -0,0 +1,133 @@
+            else:
+                self.log_group_id = log_groups[0].id
+
+            list_logs_response = self.logging_client.list_logs(
+                log_group_id=self.log_group_id, display_name=self.log_stream
+            )
+            if not list_logs_response.data:
+                self.log_stream_id = self.create_log()
+            elif list_logs_response.data and list_logs_response.data[0].lifecycle_state != 'ACTIVE':
+                raise ValueError(
+                    f"Log stream {list_logs_response.data[0].display_name} is not ACTIVE"
+                )
+            else:
+                self.log_stream_id = list_logs_response.data[0].id
+
+        except ServiceError as se:
+            log.exception(se, stack_info=True)
+            raise ValueError(f"Unable to instantiate OCI logging handler: {se.message}")
+        except ValueError as ve:
+            raise ve
+        except Exception as e:
+            log.exception(e, stack_info=True)
+            raise ValueError(
+                "Unable to instantiate OCI logging handler. Please check the logs for error"
+            )
+
+    def create_log_group(self):
+        composite_client = oci.logging.LoggingManagementClientCompositeOperations(
+            self.logging_client
+        )
+        response = composite_client.create_log_group_and_wait_for_state(
+            create_log_group_details=oci.logging.models.CreateLogGroupDetails(
+                compartment_id=self.log_compartment_id,
+                display_name=self.log_group,
+                description="Cloud Custodian Logs",
+            ),
+            wait_for_states=['SUCCEEDED'],
+        )
+        return response.data.resources[0].identifier
+
+    def create_log(self):
+        composite_client = oci.logging.LoggingManagementClientCompositeOperations(
+            self.logging_client
+        )
+        response = composite_client.create_log_and_wait_for_state(
+            log_group_id=self.log_group_id,
+            create_log_details=oci.logging.models.CreateLogDetails(
+                display_name=self.log_stream,
+                is_enabled=True,
+                log_type="CUSTOM",
+                configuration=oci.logging.models.Configuration(
+                    source=oci.logging.models.Source(source_type="UNKNOWN_ENUM_VALUE"),
+                    compartment_id=self.log_compartment_id,
+                ),
+            ),
+            wait_for_states=['SUCCEEDED'],
+        )
+        return response.data.resources[0].identifier
+
+    def emit(self, message):
+        msg = self.format_message(message)
+        tz = datetime.now().astimezone().tzinfo
+        try:
+            self.loggingingestion_client.put_logs(
+                log_id=self.log_stream_id,
+                put_logs_details=oci.loggingingestion.models.PutLogsDetails(
+                    specversion=importlib.metadata.version("c7n_oci"),
```

Comment:
move this to a module/global level variable/constant

Reply:
Moved to c7n_oci/utils